### PR TITLE
Adding ARM template that uses CustomScriptExtension to install Vert.x, OpenJDK, Apache, and MySQL Server on Ubuntu Server

### PR DESCRIPTION
**Pull request by OguzPastirmaci**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@OguzPastirmaci__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by OguzPastirmaci** [comment]
The email on Friday: "No azuredeploy-parameters.json file was found in the expected location"

So I renamed azuredeploy.parameters.json to "azuredeploy-parameters.json" and now I received an email: "No azuredeploy.parameters.json file was found in the expected location."

What is the correct format?

**Comment by singhkay** [comment]
@OguzPastirmaci We fixed the bot. It now looks for azuredeploy.parameters.json. This was discussed in #420

File: vertx-openjdk-apache-mysql-on-ubuntu/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  "parameters": {
+    "newStorageAccountName": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the Storage Account where the Virtual Machine's disks will be placed."
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "East US",
```
Comment (singhkay) [inline vertx-openjdk-apache-mysql-on-ubuntu/azuredeploy.json]: Please remove the default value and let the user pick the location

File: vertx-openjdk-apache-mysql-on-ubuntu/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+      "metadata": {
+        "description": "The Ubuntu version for the VM. This will pick a fully patched image of this given Ubuntu version. Default value is 14.04.2-LTS"
+      }
+    }
+  },
+  "variables": {
+    "imagePublisher": "Canonical",
+    "imageOffer": "UbuntuServer",
+    "OSDiskName": "osdiskforlinuxsimple",
+    "nicName": "myVMNic",
```
Comment (singhkay) [inline vertx-openjdk-apache-mysql-on-ubuntu/azuredeploy.json]: Please rename resources like this to use this scheme <templateScenario><resourceName>

File: vertx-openjdk-apache-mysql-on-ubuntu/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+      "defaultValue": "14.04.2-LTS",
+      "metadata": {
+        "description": "The Ubuntu version for the VM. This will pick a fully patched image of this given Ubuntu version. Default value is 14.04.2-LTS"
+      }
+    }
+  },
+  "variables": {
+    "imagePublisher": "Canonical",
+    "imageOffer": "UbuntuServer",
+    "OSDiskName": "osdiskforlinuxsimple",
```
Comment (singhkay) [inline vertx-openjdk-apache-mysql-on-ubuntu/azuredeploy.json]: Rename this as well to <templateScenario><resourceName>

File: vertx-openjdk-apache-mysql-on-ubuntu/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    "nicName": "myVMNic",
+    "addressPrefix": "10.0.0.0/16",
+    "subnetName": "Subnet",
```
Comment (singhkay) [inline vertx-openjdk-apache-mysql-on-ubuntu/azuredeploy.json]: Rename

File: vertx-openjdk-apache-mysql-on-ubuntu/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    "subnetPrefix": "10.0.0.0/24",
+    "storageAccountType": "Standard_LRS",
+    "publicIPAddressName": "myPublicIP",
```
Comment (singhkay) [inline vertx-openjdk-apache-mysql-on-ubuntu/azuredeploy.json]: Rename

File: vertx-openjdk-apache-mysql-on-ubuntu/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+    "publicIPAddressType": "Dynamic",
+    "vmStorageAccountContainerName": "vhds",
+    "vmSize": "Standard_A2",
+    "virtualNetworkName": "MyVNET",
```
Comment (singhkay) [inline vertx-openjdk-apache-mysql-on-ubuntu/azuredeploy.json]: Rename

File: vertx-openjdk-apache-mysql-on-ubuntu/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[parameters('newStorageAccountName')]",
+      "apiVersion": "2015-05-01-preview",
```
Comment (singhkay) [inline vertx-openjdk-apache-mysql-on-ubuntu/azuredeploy.json]: it's a good idea to make the API version a variable so it's easier to move to newer version headers

File: vertx-openjdk-apache-mysql-on-ubuntu/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+      "dependsOn": [
+        "[concat('Microsoft.Compute/virtualMachines/', parameters('vmName'))]"
+      ],
+      "properties": {
+        "publisher": "Microsoft.OSTCExtensions",
+        "type": "CustomScriptForLinux",
+        "typeHandlerVersion": "1.3",
+        "settings": {
+          "fileUris": [
+            "https://raw.githubusercontent.com/OguzPastirmaci/azure-quickstart-templates/master/vertx-openjdk-apache-mysql-on-ubuntu/install.sh"
```
Comment (singhkay) [inline vertx-openjdk-apache-mysql-on-ubuntu/azuredeploy.json]: This script should be located in the template folder with this template

File: vertx-openjdk-apache-mysql-on-ubuntu/readme.md [hunk 9]
```diff
@@ -0,0 +1,4 @@
+# Vert.x, OpenJDK, Apache, and MySQL Server on Ubuntu VM
+
+This template uses the Azure Linux CustomScript extension to deploy Vert.x, OpenJDK, Apache, and MySQL Server on Ubuntu 14.04 LTS.
```
Comment (singhkay) [inline vertx-openjdk-apache-mysql-on-ubuntu/readme.md]: Probably a good idea to explain a bit more about Vert.x and the scenario this template accomplishes and what it doesn't do

**Comment by OguzPastirmaci** [comment]
Made the changes according to your comments.

**Comment by OguzPastirmaci** [comment]
How do I fix the issue below? I checked but couldn't find the cause.

•The file azuredeploy.json is not valid. Response from ARM API: BadRequest - {"error":{"code":"InvalidTemplate","message":"Deployment template validation failed: 'The value for the template parameter 'adminPassword' at line '35' and column '23' is not provided.'."}}

**Comment by singhkay** [comment]
@OguzPastirmaci Looks like your parameters file is missing some of the parameters in the template.

**Comment by singhkay** [comment]
@OguzPastirmaci Looks good! Thank you for contributing. Welcome to the repo! merged

**Pull request merged**